contracts/treasury/LivepeerGovernor.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/GovernorUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorSettingsUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorTimelockControlUpgradeable.sol";
+
+import "../bonding/libraries/EarningsPool.sol";
+import "../bonding/libraries/EarningsPoolLIP36.sol";
+import "../polling/PollCreator.sol";
+
+import "../ManagerProxyTarget.sol";
+import "../IController.sol";
+import "../rounds/IRoundsManager.sol";
+import "./GovernorCountingOverridable.sol";
+import "./BondingCheckpointsVotes.sol";
+
+/**
+ * @title LivepeerGovernor
+ * @notice Core contract for Livepeer governance, starting as the treasury governor.
+ * @dev If we ever add fields to this class or more extensions, make sure to add a storage gap to our custom
+ * GovernorCountingOverridable extension.
+ */
+contract LivepeerGovernor is
+    Initializable,
+    ManagerProxyTarget,
+    GovernorUpgradeable,
+    GovernorSettingsUpgradeable,
+    GovernorTimelockControlUpgradeable,
+    GovernorVotesUpgradeable,
+    GovernorVotesQuorumFractionUpgradeable,
+    GovernorCountingOverridable
+{
+    /**
+     * @notice TreasuryGovernor constructor. Only invokes constructor of base Manager contract with provided Controller.
+     * @dev This constructor will not initialize any state variables besides `controller`. The `initialize` function
+     * must be called through the proxy after construction to initialize the contract's state in the proxy contract.
+     * @param _controller Address of Controller that this contract will be registered with
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address _controller) Manager(_controller) {
```

Comment:
So this is an existing code pattern in our protocol, since I assume OZ's proxy didn't even exist back when we wrote these. For general context I think this is the best documentation:
https://github.com/livepeer/protocol/blob/77a37e58192a6017a200920f58551d2cfc2873a3/contracts/ManagerProxy.sol#L7
(can go up on `ManagerProxyTarget` as well)

The core differences from OZ proxy, from my understanding, are:
1. It doesn't have a fixed target, but it fetches it from the `Controller` instead which holds an association of "contract id -> address/other info"
2. Thus it also contains some storage on its own, to hold: the `controllerAddress` and the `targetContractId`

The TLDR is that targets then need to also inherit from the `ManagerProxyTarget` storage layout otherwise they would mismatch. The constructor comes as an obligation of inheriting from `Manager`, but in practice the storage variables initialized there are never really used on the implementation contract.

We could use the pure OZ transparent proxy for this, but it would be the only contract in our protocol to use a different proxying mechanism, which would also need custom tooling only for it. So I think there are some edges to clear (as detailed in the other comment), but I think it's worth trying to stick to our existing proxy/upgradeable pattern for this.

Reply:
Resolving as per https://github.com/livepeer/protocol/pull/615#discussion_r1276844201

Reply:
Yeah, I agree that sticking to our existing proxy makes sense. It looks like you resolved the contract storage issues, which is really what I am concerned about. And, since this is only calling `_disableInitializers()`, there's no issue here.